padertorch/train/hooks.py

```diff
@@ -1022,3 +1023,46 @@ def set_value(self, trainer, value):
         optimizer = self.get_optimizer(trainer)
         for param_group in optimizer.param_groups:
             param_group['lr'] = value
+
+
+class EmissionsTrackerHook(TriggeredHook):
+    @property
+    def priority(self):
+        return Priority.SUMMARY
+
+    def __init__(self, trigger, prefix='x_emissions', storage_dir=None):
+        super().__init__(trigger)
+        self.prefix = prefix
+        self.storage_dir = storage_dir
+        from codecarbon import EmissionsTracker
+        self.tracker = EmissionsTracker(
+            output_dir=storage_dir, on_csv_write="update", log_level='error')
+        self.tracker.start()
+
+    def pre_step(self, trainer: 'pt.Trainer'):
+        if self.trigger(iteration=trainer.iteration, epoch=trainer.epoch):
+            self.tracker.flush()
+            self.dump_emissions(trainer)
+
+    def dump_emissions(self, trainer):
+        emissions_file = Path(self.storage_dir) / 'emissions.csv'
+        with emissions_file.open() as fid:
+            lines = fid.read().split('\n')
```

Comment:
Is it possible to read the data from the tracker object directly? If so, I would prefer to avoid a file access

Reply:
Yea I would have preferred so too, but this is only possible when calling private methods (_measure_power_and_energy and _prepare_emissions_data) when I understand correctly.

Reply:
unfortunately, the flush() method only returns emissions_data.emissions (which are the CO2 emissions in kgs) rather than all the emissions_data including gpu_energy, cpu_energy and ram_energy

Reply:
I would then stick to the official public interface